### ISSUE TITLE
AngularJS will be deprecated in Grafana

### ISSUE THREAD

**Issue by christophertjones123** [etichy/grafana-swis-datasource#28]
After upgrading to Grafana 10.1.2 I am receiving a message that this plugin will no longer work in a future release after AngularJS is deprecated. Will this plugin be converted to a supported model? As well as possibly available in the Grafana plugin catalog as well to make it easier to install? I absolutely love this plugin and what it provides and I’d hate to lose the ability to use it. If I can lend a hand I’m more than happy to, I have no idea how to make custom plugins but I’m more than happy to try and learn if it means keeping this plugin around for the community.

**Comment by jHumeLPL** [comment]
This guy built this SolarWinds plugin 5 years ago and I don't think he's updated it since (**EDIT**: _I'm wrong, it looks like he has updated it about 6 months ago to fix a problem with it.  So maybe there's hope he can migrate it off of Angular!_).  Unfortunately, we're gonna have to either use the **Infinity** Data Source for **SolarWinds** queries, hope somebody builds a new data source plugin for **SolarWinds**, or build our own Data Source plugin.

They don't plan on actually blocking **Angular** plugins until at least **Grafana 12** (which is just over a year away right now).  Until then you can set the "**angular_support_enabled**" flag to "**true**" in the "**security**" section of your Grafana **custom.ini** file (that will allow angular plugins to still work) as well as set the "**hide_angular_deprecation**" flag in the "**plugins**" section of the **custom.ini** file to "**true**" (that will hide the annoying message on any dashboards that have one or more angular plugins).

I myself was able to get the development environment for data source plugins installed on my work computer, but actually developing a plugin is currently a bit over my head.  Maybe one day I can find the time to sit down and give it a shot, but for now I'm either at the mercy of others or gonna have to just keep relying on the **Infinity** plugin...

**Comment by christophertjones123** [comment]
@seismiccollision I don’t think this is necessary anymore. In Grafana 12 (I can’t remember exactly what version) Grafana released an official data source connector for SWIS. I haven’t personally had the chance to try it out but from the documentation and the video it looks to be fairly straightforward and usable.
